sequencer/src/l1_client.rs

```diff
@@ -129,6 +130,43 @@ impl L1Client {
             }
         }
     }
+    /// Get fee info for each `Deposit` ocurring between `prev`
+    /// and `new`. Returns `Vec<FeeInfo>`
+    pub async fn _get_finalized_deposits(
+        &self,
+        prev_finalized: Option<u64>,
+        new_finalized: u64,
+        address: Address,
+    ) -> Vec<FeeInfo> {
+        // `prev` should have allready been processed unless we
+        // haven't processed *any* blocks yet.
+        let prev = prev_finalized.map(|prev| prev + 1).unwrap_or(0);
+        // No new blocks have been finalized, therefore there are no
+        // new deposits.
+        if prev == new_finalized {
```

Comment:
I think there is still an off-by-1 issue here. Imagine the previous and new finalized blocks are both 1. But we added 1 to `prev` above so we get `2 != 1` here and fail this check when it should pass

Reply:
65c6ad67a71a644d055e95d85e6f5a3233327ab6 should address it.